addons/l10n_es_edi_tbai/models/ir_attachment.py

```diff
@@ -17,21 +18,21 @@ def action_download_xsd_files(self):
             self.env, 'https://www.w3.org/TR/xmldsig-core/xmldsig-core-schema.xsd', 'xmldsig-core-schema.xsd',
             xsd_name_prefix='l10n_es_edi_tbai')
 
-        agency = self.env.company.l10n_es_tbai_tax_agency
-        urls = get_key(agency, 'xsd_url')
-        names = get_key(agency, 'xsd_name')
-        # For Bizkaia, one url per XSD (post/cancel)
-        if isinstance(urls, dict):
-            for move_type in ('post', 'cancel'):
+        for agency in ['gipuzkoa', 'araba', 'bizkaia']:
+            urls = get_key(agency, 'xsd_url')
+            names = get_key(agency, 'xsd_name')
+            # For Bizkaia, one url per XSD (post/cancel)
+            if isinstance(urls, dict):
+                for move_type in ('post', 'cancel'):
+                    xml_utils.load_xsd_files_from_url(
+                        self.env, urls[move_type], names[move_type],
+                        xsd_name_prefix='l10n_es_edi_tbai',
+                    )
+            # For other agencies, single url to zip file (only keep the desired names)
+            else:
                 xml_utils.load_xsd_files_from_url(
-                    self.env, urls[move_type], names[move_type],
+                    self.env, urls, agency + '_XSDs.zip',
```

Comment:
@jco-odoo we would have to add the agency name in the method fetching the XSD with which to validate.

Reply:
This name was discarded as ZIPs contain multiple files and one single name does not make sense in that case. Removed.